library/lua/dfhack.lua

```diff
@@ -367,46 +395,77 @@ function pos2xy(pos)
     end
 end
 
-function xy2pos(x,y)
+---@nodiscard
+---@param x number
+---@param y number
+---@return coord2d
+function xy2pos(x, y)
     if x then
-        return {x=x,y=y}
+        return { x = x, y = y }
     else
-        return {x=-30000,y=-30000}
+        return { x = -30000, y = -30000 }
     end
 end
 
+---@nodiscard
+---@param a coord|coord2d
+---@param b coord|coord2d
+---@return boolean
 function same_xy(a,b)
     return a and b and a.x == b.x and a.y == b.y
 end
 
+---@nodiscard
+---@param path coord_path|coord2d_path
+---@param i number
+---@return integer x
+---@return integer y
 function get_path_xy(path,i)
     return path.x[i], path.y[i]
 end
 
-function safe_index(obj,idx,...)
+-- Walks a sequence of dereferences, which may be represented by numbers or
+-- strings. Returns nil if any of obj or indices is nil, or a numeric index is
+-- out of array bounds.
+---@generic T
+---@param obj `T`
+---@param idx number|string
+---@param ... number|string
+---@return T obj
+function safe_index(obj, idx, ...)
     if obj == nil or idx == nil then
         return nil
     end
     if type(idx) == 'number' and
-            type(obj) == 'userdata' and -- this check is only relevant for c++
-            (idx < 0 or idx >= #obj) then
+        type(obj) == 'userdata' and     -- this check is only relevant for c++
+        (idx < 0 or idx >= #obj) then
```

Comment:
the convention I've been following here is that if the conditions are multi-line, then the `then` should go on a new line by itself

Reply:
Gotcha, is how I formatted it now good?

Reply:
looks good. part of the problem was just that the formatting changes were making the PR more difficult to grok